### PR TITLE
HDDS-1396 : Recon start fails due to changes in Aggregate Schema definition.

### PR DESCRIPTION
**Pull request by avijayanhwx**
- bin/ozone looks for hadoop-ozone-recon-*.jar as the main JAR for Recon Server. Since the codegen jar (hadoop-ozone-recon-codegen) also satisfies the above, it gets picked up as the main JAR and recon start fails due to CNF issue. To fix this, we have changed artifact name of codegen module such that it does not get picked up as Application JAR for recon. 
- Fixed guice issue for DataSourceConfiguration which was due to double binding of the same class. 
- Downgraded guice version to 4.0, to keep it compatible to dependencies.

### CODE REVIEW

**Comment by hadoop-yetus** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| 0 | reexec | 25 | Docker mode activated. |
||| _ Prechecks _ |
| +1 | @author | 0 | The patch does not contain any @author tags. |
| +1 | test4tests | 0 | The patch appears to include 1 new or modified test files. |
||| _ trunk Compile Tests _ |
| 0 | mvndep | 38 | Maven dependency ordering for branch |
| +1 | mvninstall | 998 | trunk passed |
| +1 | compile | 106 | trunk passed |
| +1 | checkstyle | 43 | trunk passed |
| +1 | mvnsite | 160 | trunk passed |
| +1 | shadedclient | 797 | branch has no errors when building and testing our client artifacts. |
| 0 | findbugs | 0 | Skipped patched modules with no Java source: hadoop-ozone |
| +1 | findbugs | 66 | trunk passed |
| +1 | javadoc | 103 | trunk passed |
||| _ Patch Compile Tests _ |
| 0 | mvndep | 15 | Maven dependency ordering for patch |
| +1 | mvninstall | 333 | the patch passed |
| +1 | compile | 112 | the patch passed |
| +1 | javac | 112 | the patch passed |
| +1 | checkstyle | 25 | the patch passed |
| +1 | mvnsite | 157 | the patch passed |
| +1 | whitespace | 0 | The patch has no whitespace issues. |
| +1 | xml | 4 | The patch has no ill-formed XML file. |
| +1 | shadedclient | 759 | patch has no errors when building and testing our client artifacts. |
| 0 | findbugs | 0 | Skipped patched modules with no Java source: hadoop-ozone |
| +1 | findbugs | 83 | the patch passed |
| +1 | javadoc | 61 | hadoop-ozone generated 0 new + 2 unchanged - 14 fixed = 2 total (was 16) |
| +1 | javadoc | 19 | ozone-recon-codegen in the patch passed. |
| +1 | javadoc | 20 | ozone-recon in the patch passed. |
||| _ Other Tests _ |
| -1 | unit | 1439 | hadoop-ozone in the patch failed. |
| +1 | unit | 27 | ozone-recon-codegen in the patch passed. |
| +1 | unit | 45 | ozone-recon in the patch passed. |
| +1 | asflicense | 38 | The patch does not generate ASF License warnings. |
| | | 5421 | |


| Reason | Tests |
|-------:|:------|
| Failed junit tests | hadoop.ozone.client.rpc.TestOzoneRpcClientWithRatis |
|   | hadoop.ozone.client.rpc.TestOzoneRpcClient |
|   | hadoop.ozone.client.rpc.TestBlockOutputStream |
|   | hadoop.ozone.om.TestScmChillMode |
|   | hadoop.ozone.TestMiniChaosOzoneCluster |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=17.05.0-ce Server=17.05.0-ce base: https://builds.apache.org/job/hadoop-multibranch/job/PR-700/1/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hadoop/pull/700 |
| Optional Tests |  dupname  asflicense  compile  javac  javadoc  mvninstall  mvnsite  unit  shadedclient  xml  findbugs  checkstyle  |
| uname | Linux 68537c1bd0d6 4.4.0-138-generic #164-Ubuntu SMP Tue Oct 2 17:16:02 UTC 2018 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | personality/hadoop.sh |
| git revision | trunk / 77fe51e |
| maven | version: Apache Maven 3.3.9 |
| Default Java | 1.8.0_191 |
| findbugs | v3.1.0-RC1 |
| unit | https://builds.apache.org/job/hadoop-multibranch/job/PR-700/1/artifact/out/patch-unit-hadoop-ozone.txt |
|  Test Results | https://builds.apache.org/job/hadoop-multibranch/job/PR-700/1/testReport/ |
| Max. process+thread count | 5359 (vs. ulimit of 5500) |
| modules | C: hadoop-ozone hadoop-ozone/ozone-recon-codegen hadoop-ozone/ozone-recon U: hadoop-ozone |
| Console output | https://builds.apache.org/job/hadoop-multibranch/job/PR-700/1/console |
| Powered by | Apache Yetus 0.9.0 http://yetus.apache.org |


This message was automatically generated.

**Comment by vivekratnavel** [comment]
+1 LGTM

File: hadoop-ozone/ozone-recon-codegen/pom.xml [hunk 1]
```diff
@@ -21,11 +21,10 @@
     <version>0.5.0-SNAPSHOT</version>
   </parent>
   <modelVersion>4.0.0</modelVersion>
-  <artifactId>hadoop-ozone-recon-codegen</artifactId>
+  <artifactId>hadoop-ozone-reconcodegen</artifactId>
```
Comment (bharatviswa504) [inline hadoop-ozone/ozone-recon-codegen/pom.xml]: Why this naming is changed?
Now it is different from the project directory structure?

Comment (bharatviswa504) [reply hadoop-ozone/ozone-recon-codegen/pom.xml]: Thanks got info from PR description.

**Comment by bharatviswa504** [comment]
Do we need to rebase this branch, as acceptance tests are passing locally?

I don't think failure is related, can we rebase the branch to get a clean run?

**Comment by hadoop-yetus** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| 0 | reexec | 39 | Docker mode activated. |
||| _ Prechecks _ |
| +1 | @author | 0 | The patch does not contain any @author tags. |
| +1 | test4tests | 0 | The patch appears to include 1 new or modified test files. |
||| _ trunk Compile Tests _ |
| 0 | mvndep | 48 | Maven dependency ordering for branch |
| +1 | mvninstall | 1122 | trunk passed |
| +1 | compile | 129 | trunk passed |
| +1 | checkstyle | 101 | trunk passed |
| +1 | mvnsite | 192 | trunk passed |
| +1 | shadedclient | 995 | branch has no errors when building and testing our client artifacts. |
| 0 | findbugs | 0 | Skipped patched modules with no Java source: hadoop-ozone |
| +1 | findbugs | 77 | trunk passed |
| +1 | javadoc | 110 | trunk passed |
||| _ Patch Compile Tests _ |
| 0 | mvndep | 14 | Maven dependency ordering for patch |
| +1 | mvninstall | 357 | the patch passed |
| +1 | compile | 119 | the patch passed |
| +1 | javac | 119 | the patch passed |
| +1 | checkstyle | 25 | the patch passed |
| +1 | mvnsite | 164 | the patch passed |
| +1 | whitespace | 0 | The patch has no whitespace issues. |
| +1 | xml | 4 | The patch has no ill-formed XML file. |
| +1 | shadedclient | 822 | patch has no errors when building and testing our client artifacts. |
| 0 | findbugs | 0 | Skipped patched modules with no Java source: hadoop-ozone |
| +1 | findbugs | 85 | the patch passed |
| +1 | javadoc | 98 | the patch passed |
||| _ Other Tests _ |
| -1 | unit | 1629 | hadoop-ozone in the patch failed. |
| +1 | unit | 29 | ozone-recon-codegen in the patch passed. |
| +1 | unit | 47 | ozone-recon in the patch passed. |
| +1 | asflicense | 34 | The patch does not generate ASF License warnings. |
| | | 6091 | |


| Reason | Tests |
|-------:|:------|
| Failed junit tests | hadoop.ozone.scm.TestAllocateContainer |
|   | hadoop.ozone.client.rpc.TestCommitWatcher |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=17.05.0-ce Server=17.05.0-ce base: https://builds.apache.org/job/hadoop-multibranch/job/PR-700/2/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hadoop/pull/700 |
| Optional Tests |  dupname  asflicense  compile  javac  javadoc  mvninstall  mvnsite  unit  shadedclient  xml  findbugs  checkstyle  |
| uname | Linux 747e2b996f22 4.4.0-141-generic #167~14.04.1-Ubuntu SMP Mon Dec 10 13:20:24 UTC 2018 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | personality/hadoop.sh |
| git revision | trunk / 7fa73fa |
| maven | version: Apache Maven 3.3.9 |
| Default Java | 1.8.0_191 |
| findbugs | v3.1.0-RC1 |
| unit | https://builds.apache.org/job/hadoop-multibranch/job/PR-700/2/artifact/out/patch-unit-hadoop-ozone.txt |
|  Test Results | https://builds.apache.org/job/hadoop-multibranch/job/PR-700/2/testReport/ |
| Max. process+thread count | 5011 (vs. ulimit of 5500) |
| modules | C: hadoop-ozone hadoop-ozone/ozone-recon-codegen hadoop-ozone/ozone-recon U: hadoop-ozone |
| Console output | https://builds.apache.org/job/hadoop-multibranch/job/PR-700/2/console |
| Powered by | Apache Yetus 0.9.0 http://yetus.apache.org |


This message was automatically generated.

**Comment by bharatviswa504** [comment]
Test failures are not related to this patch.
I will commit this.

**Pull request merged**